include/bluetooth/audio/audio.h

```diff
@@ -1745,14 +1776,125 @@ int bt_audio_broadcast_sink_stop(struct bt_audio_broadcast_sink *sink);
+
+/** @brief Extract frame duration from BT codec config
+ *
+ *  @param codec The codec configuration to extract data from.
+ *
+ *  @return Frame duration in microseconds if value is found else a negative value
+ *          of type @ref bt_audio_codec_parse_err.
+ */
+int bt_codec_cfg_get_frame_duration_us(const struct bt_codec *codec);
+
+/** @brief Extract channel allocation from BT codec config
+ *
+ * The value returned is a bit field representing one or more audio locations as
+ * specified by @ref bt_audio_location
+ * Shall match one or more of the bits set in BT_PAC_SNK_LOC/BT_PAC_SRC_LOC.
+ *
+ * Up to the configured @ref BT_CODEC_LC3_CHAN_COUNT number of channels can be present.
+ *
+ *  @param codec The codec configuration to extract data from.
+ *  @param chan_allocation Pointer to the variable to store the extracted value in.
+ *
+ *  @return BT_AUDIO_CODEC_PARSE_SUCCESS if value is found and stored in the pointer provided
+ *          else a negative value of type @ref bt_audio_codec_parse_err.
+ */
+int bt_codec_cfg_get_chan_allocation_val(const struct bt_codec *codec, uint32_t *chan_allocation);
+
+/** @brief Extract frame size in octets from BT codec config
+ *
+ * The overall SDU size will be octets_per_frame * blocks_per_sdu.
+ *
+ * The Bluetooth specificationa are not clear about this value - it does not state that
+ * the codec shall use this SDU size only. A codec like LC3 supports variable bit-rate
+ * (per SDU) hence it might be allowed for an encoder to reduce the frame size below this
+ * value.
+ * Hence it is recommended to use the received SDU size and divide by
+ * blocks_per_sdu rather than relying on this octets_per_sdu value to be fixed.
+ *
+ *  @param codec The codec configuration to extract data from.
+ *
+ *  @return Frame length in octets if value is found else a negative value
+ *          of type @ref bt_audio_codec_parse_err.
+ */
+int bt_codec_cfg_get_octets_per_frame(const struct bt_codec *codec);
+
+/** @brief Extract number of audio frames in each SDU from BT codec config
+ *
+ * The overall SDU size will be octets_per_frame * blocks_per_sdu.
+ *
+ * If this value is not present a default value of 1 shall be used.
+ *
+ * The Bluetooth specification names the frames in the codec configuration blocks.
+ *
+ *  @param codec The codec configuration to extract data from.
+ *  @param fallback_to_default If true this function will return the default value of 1
+ *         if the type is not found.
+ *
+ *  @return The count of codec frames in each SDU if value is found else a negative value
+ *          of type @ref bt_audio_codec_parse_err - unless when \p fallback_to_default is true
+ *          then the value 1 is returned if frames per sdu is not found.
+ */
+int bt_codec_cfg_get_frames_per_sdu(const struct bt_codec *codec, bool fallback_to_default);
+
+/**
+ * @brief Lookup a specific value based on type
+ *
+ * @param codec The codec data to search in.
+ * @param type The type id to look for
```

Comment:
Do we have these types defined? If so, should add a reference. If not, we probably should, right?

Reply:
We do, but they are scattered over lc3.h and the only common about their name is BT_CODEC - hence not much help there, and the metadata types are in audio.h...
Should be fixed when cleanup of the lc3.h is done (#44342). 
Best option a user have is to look into what is put into a `bt_codec` data type - based on the context it will be either codec config, codec capability or metadata.